sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2323,6 +2324,7 @@ CurlConnection::CurlConnection(
           + std::string(curl_easy_strerror(result)));
     }
   }
+#endif
```

Comment:
We should add an #else if `(!options.SslOptions.PemEncodedExpectedRootCertificates.empty())`, throw an exception for option not supported on this version of curl. This will let the called know this option won't work for them, and becomes self-documenting at runtime.

Reply:
I don't see any reason to throw at runtime.  You know at link time if you are on the right version.  I'm leaning towards Larry's suggestion to only add the value in the options if the option is present.